### DISCUSSION TITLE
Cannot login Copilot on IntelliJ IDEA under Jetbrains Gateway

### DISCUSSION THREAD

**Discussion by ganadist** [community/community#8333] [Copilot Conversations] [answered]
`Jetbrains Gateway` is remote development solution
https://www.jetbrains.com/remote-development/gateway/

And `Github Copilot` plugin can be installed on IntelliJ IDEA under Gateway.

But when I tried to login copilot, plugin cannot open browser, and login process went to blocked.

X-posted at https://youtrack.jetbrains.com/issue/GTW-742

**Comment by jansorg** [comment, marked as answer]
Please try to manually open the website and to enter the code. Both code and link is shown in the login dialog. You don't have to enter the code on the same machine. Click "Ok" instead of "Copy and Open" to avoid the automatic opening in a browser.

**Comment by hirshira** [comment]
cannot login

**Comment by sanderha** [comment]
Logging in seems to work fine now, but suggestions dont seem to work, even though the loading icon is spinning.. 
Has anyone actually had success using copilot via gateway?

**Reply by cosine0** [reply]
Now it shows the suggestion in line, but I can't accept the suggestion to add it to the code. Double-shift action "Copilot: Apply completion to Editor" or adding a keymap won't work.
A workaround to use it is to open the GitHub Copilot side panel and click on one in the suggestion list.

**Comment by elisim** [comment]
Happy new year.
Any new workarounds? seems that it's very hard to use CoPilot with Jetbrains Gateway on Codespaces.

**Comment by silostack** [comment]
any update on this? also getting this issue. seems like a copilot issue though cause i can't get it to work with visual code either

**Comment by DavidGodefroid** [comment]
It took me a long time to find it because the login window wouldn't open. First I had to click on "logout from Github" (in Tools > Github Copilot). After that, clicking on login, the window opened.

**Reply by patzu** [reply]
Man! The problem is not finding the login window?! The problem is Intellij getting connection request after authentication and do not react as connected to Copilot! I stuck on that and tried many solutions but none of them worked. Any new approaches are welcomed.

**Reply by gschurck** [reply]
Strange but for me logging out and logging in again fixed the issue...

**Comment by igor-zp** [comment]
Have same issue
<img width="659" alt="image" src="https://user-images.githubusercontent.com/35780695/216273011-e915dffe-e2ef-4a37-9655-9e8cabd26ee4.png">
<img width="261" alt="image" src="https://user-images.githubusercontent.com/35780695/216273073-a817aa6e-d5ca-480f-b880-68ac83483e89.png">

**Reply by patzu** [reply]
Yeap, still same problem here as well.

**Reply by Unrasend** [reply]
also stuck on that

**Reply by topolanekmartin** [reply]
I had the same problem and it helped to check copilot on github. I had an expired trial and had to get a premium one.

**Reply by cryptothink629** [reply]
same here, any solution now? thank you

**Reply by thaipham98** [reply]
I had the same problem and got it fixed. I need to signup Copilot for my git account before the JetBrains authentication.

Signup guide here:
https://docs.github.com/en/copilot/quickstart

**Reply by michalstocki** [reply]
I have the same problem having the Githu copilot enabled for a github organization. Is it required to have a personal copilot enabled?

**Comment by steve-cuzzort** [comment]
Also cannot login to copilot in webstorm.

**Comment by guzmle2** [comment]
I had the same problem, I solved it by going to Github > My profile > Copilot, authorize the use of copilot, update my data, and then go back to jetbrains and repeat the steps

**Comment by gschurck** [comment]
Logging out of Github Copilot and then log in again fixed the issue for me.

**Comment by andrii0lomakin** [comment]
Same for me. Just log out of Git Hub and log in during the authorization request. That solves this problem.

**Comment by alangpierce** [comment]
I also ran into problems logging in to GitHub in JetBrains Gateway. In my case, I clicked the login button but the window didn't open at all. My solution was to install Copilot as a client plugin rather than a host plugin, see settings screenshot below of the settings window.

<img width="290" alt="Screenshot 2023-11-13 at 11 48 53 PM" src="https://github.com/community/community/assets/211605/8992b8f4-163c-4f03-b240-1f07af92e69b">

With the host plugin, the logs showed an exception `The showAndGet() method is for modal dialogs only`, which is also mentioned in this other discussion: https://github.com/orgs/community/discussions/68781

**Comment by rajathbrgowda** [comment]
I had a similar issue but it was resolved after removing the installed plugin, reinstalling, and restarting the IDE. Additionally, I ensured, I logged in via the copilot icon on the bottom right corner

**Comment by OzanKurt** [comment]
https://github.com/orgs/community/discussions/46917 Still doesn't work
